scheduler/pkg/agent/repository/mlserver/settings.go

```diff
@@ -0,0 +1,182 @@
+		case "lightgbm":
+			return createLightGBMModelSettings(path)
+		case "mlflow":
+			return createMLFlowModelSettings()
+		case "python":
+			return createCustomPythonModelSettings(path)
+		case "sklearn":
+			return createSKLearnModelSettings(path)
+		case "xgboost":
+			return createXGBoostModelSettings(path)
+		}
+	}
+	return nil, fmt.Errorf("Can't create model-settings from requirements %v", modelSpec.Requirements)
+}
+
+func saveModelSettings(path string, modelSettings *ModelSettings) error {
+	data, err := json.Marshal(modelSettings)
+	if err != nil {
+		return err
+	}
+	settingsPath := filepath.Join(path, mlserverConfigFilename)
+	return os.WriteFile(settingsPath, data, fs.ModePerm)
+}
+
+func findFilesMatchingExtension(path string, ext string) ([]string, error) {
+	return filepath.Glob(fmt.Sprintf("%s/*.%s", path, ext))
+}
+
+func findModelUri(path string, extensions []string) (string, error) {
+	modelUri := ""
+	for _, ext := range extensions {
+		matches, err := findFilesMatchingExtension(path, ext)
+		if err != nil {
+			return "", err
+		}
+		if matches != nil {
+			modelUri = matches[0]
+			break
+		}
+	}
+	modelUri = strings.TrimPrefix(modelUri, path)
+	if modelUri == "" {
+		return "", fmt.Errorf("Failed to find sklearn artifact in %s", path)
+	}
+	return modelUri, nil
+}
+
+func createModelSettingsFromUri(modelUri string, implementation string) *ModelSettings {
+	return &ModelSettings{
+		Implementation: implementation,
+		Parameters: &ModelParameters{
+			Uri: fmt.Sprintf(".%s", modelUri),
+		},
+	}
+}
+
+func createSKLearnModelSettings(path string) (*ModelSettings, error) {
+	modelUri, err := findModelUri(path, sklearnExtensions)
+	if err != nil {
+		return nil, err
+	}
+	return createModelSettingsFromUri(modelUri, "mlserver_sklearn.SKLearnModel"), nil
+}
+
+func createXGBoostModelSettings(path string) (*ModelSettings, error) {
+	modelUri, err := findModelUri(path, xgboostExtensions)
+	if err != nil {
+		return nil, err
+	}
+	return createModelSettingsFromUri(modelUri, "mlserver_xgboost.XGBoostModel"), nil
+}
+
+func createLightGBMModelSettings(path string) (*ModelSettings, error) {
+	modelUri, err := findModelUri(path, lightgbmExtensions)
+	if err != nil {
+		return nil, err
+	}
+	return createModelSettingsFromUri(modelUri, "mlserver_lightgbm.LightGBMModel"), nil
+}
+
+func createAlibiDetectModelSettings() (*ModelSettings, error) {
+	return &ModelSettings{
+		Implementation: "mlserver_alibi_detect.AlibiDetectRuntime",
+		Parameters: &ModelParameters{
+			Uri: "./",
+		},
+	}, nil
+}
+
+func createMLFlowModelSettings() (*ModelSettings, error) {
+	return &ModelSettings{
+		Implementation: "mlserver_mlflow.MLflowRuntime",
+		Parameters: &ModelParameters{
+			Uri: "./",
+		},
+	}, nil
+}
+
+func createAlibiExplainModelSettings(path string) (*ModelSettings, error) {
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range files {
+		if f.IsDir() {
+			parallelWorkers := 0
```

Comment:
perhaps add a note in code on why parallel workers is always set to 0 for explainers?

Reply:
ok will do